In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data=pd.read_excel('C:/Users/Vishal/Desktop/LU aiml/Data/Linear Regression.xlsx')
data.head()

price  sqft_living  bedrooms  bathrooms  floors
0  221900         1180         3       1.00     1.0
1  538000         2570         3       2.25     2.0
2  180000          770         2       1.00     1.0
3  604000         1960         4       3.00     1.0
4  510000         1680         3       2.00     1.0

In [3]:
x=data[['sqft_living','bedrooms','bathrooms','floors']]
y=data[['price']]

In [7]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.25)

In [8]:
from sklearn.linear_model import LinearRegression
lr=LinearRegression()

In [9]:
lr.fit(x_train,y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [17]:
lr.coef_

array([[   309.07204755, -55901.65092153,   7906.77584916,
         -1154.78983859]])

In [18]:
lr.intercept_

array([70793.85929049])

__Equation is:__

__price = 70793.85929049 + 309.07204755(sqft_living) -55901.65092153(bedrooms) + 7906.77584916(bathrooms) -1154.78983859(floors)__

In [19]:
ypred=lr.predict(x_test)
ypred

array([[1473316.67868133],
       [ 492621.67683218],
       [ 231275.82199045],
       ...,
       [ 569328.88396844],
       [ 547734.60396517],
       [ 417866.99050043]])

In [20]:
from sklearn.metrics import r2_score,mean_squared_error
r2_score(y_test,ypred)

0.5252907383742016

In [21]:
mean_squared_error(y_test,ypred)

63676648302.16815

# INFERENCE:
R-square value is 0.525290738374 means there is 52.5% variance in the target variable.

Totat Error in the model is 63676648302.16815